src/main/java/duke/notes/event/Event.java

```diff
@@ -0,0 +1,300 @@
+            eventsOutstanding--;
+            eventsCompleted++;
+            this.printList();
+            return true;
+        } else {
+            this.printList();
+            return false;
+        }
+    }
+
+    /**
+     * This method is used to reset the static variables of the {@code Event}
+     * class in the event of a program reset.
+     */
+    public static void resetStaticVariables() {
+        eventsOutstanding = 0;
+        eventsCompleted = 0;
+    }
+
+    /**
+     * This method is used to delete an {@code Event} object.
+     * Class-level members will be updated to reflect the change in completion status.
+     */
+    public void deleteExistingNote() {
+        if(isDone){
+            System.out.print("    Event #" + this.serialNum + " was already done!");
+            System.out.println("\t...deleting the event anyway.");
+            eventsCompleted--;
+        } else {
+            System.out.println("    Noted! I've deleted Event #" + this.serialNum + ".");
+            eventsOutstanding--;
+        }
+    }
+
+    /**
+     * This method is used to set or edit the start date and time of the {@code Event} object.
+     *
+     * @param startDate The start date and time of the {@code Event}.
+     * @exception DateException If there are errors in the formats or substance of {@code Date} objects.
+     */
+    public void setStartDate(Date startDate) throws DateException {
+        Date now = new Date();
+        if(startDate.before(now)){
+            throw new DateException(startDate, "StartB4Now");
+        } else {
+            if(this.endDate != null) {
+                if(startDate.after(this.endDate)) {
+                    throw new DateException(startDate, "StartAFEnd");
+                } else {
+                    this.startDate = startDate;
+                    this.durationMinutes = (this.endDate.getTime() - startDate.getTime()) / 60000;
+                }
+            } else {
+                this.startDate = startDate;
+            }
+        }
+    }
+
+    /**
+     * This method is used to set or edit the end date and time of the {@code Event} object.
+     *
+     * @param endDate The end date and time of the {@code Event}.
+     * @exception DateException If there are errors in the formats or substance of {@code Date} objects.
+     */
+    public void setEndDate(Date endDate) throws DateException {
+        Date now = new Date();
+        if(this.startDate == null){
+            throw new DateException(endDate, "NoStartDate");
+        } else {
+            if(endDate.before(this.startDate)){
+                throw new DateException(endDate, "EndB4Start");
+            } else if(endDate.before(now)) {
+                throw new DateException(endDate, "EndB4Now");
+            } else {
+                this.endDate = endDate;
+                this.durationMinutes = (endDate.getTime() - this.startDate.getTime()) / 60000;
+            }
+        }
+    }
```

Comment:
No need for `else` as the `throw` exits the method anyway.

```java
    public void setEndDate(Date endDate) throws DateException {
        Date now = new Date();
        if(this.startDate == null){
            throw new DateException(endDate, "NoStartDate");
        } 
        if(endDate.before(this.startDate)){
            // ...
        }
    }
```

Reply:
Noted Prof, I've also looked thru the rest of the code for similar instances.